### PR TITLE
Factor groovy out of core into lang-groovy

### PR DESCRIPTION
**Pull request by rmuir**
See https://github.com/elastic/elasticsearch/issues/13725 for the motivation.

Note that many core tests (approximately 50) depend on groovy. Rather than try to refactor all of these at once, I simply moved them all to this plugin under the `org.elasticsearch.messy.tests` package, so we can deal with them iteratively. This preserves them in their entirety, so we don't lose coverage, or go back and forth on how each one should be fixed: instead this way we can clean up over time.

Last time when factoring out expressions, @rjernst stayed up all night to fix a bunch of tests, but we can't do things this way, that is why I did what I did. 

The package-info.java has guidelines about what I think we should do, to get them back in core. It also has the list of renames so you know where each file came from (in case its easy to fix and move back to core). 

I made no security changes here to not bury in this noise.

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/ExceptionSerializationTests.java [hunk 1]
```diff
@@ -758,7 +757,7 @@ public void testIds() {
         ids.put(92, org.elasticsearch.indices.recovery.DelayRecoveryException.class);
         ids.put(93, org.elasticsearch.search.warmer.IndexWarmerMissingException.class);
         ids.put(94, org.elasticsearch.client.transport.NoNodeAvailableException.class);
-        ids.put(95, org.elasticsearch.script.groovy.GroovyScriptCompilationException.class);
+        ids.put(95, null);
```
Comment (jdconrad) [inline core/src/test/java/org/elasticsearch/ExceptionSerializationTests.java]: Should this really be null?

Comment (rmuir) [inline core/src/test/java/org/elasticsearch/ExceptionSerializationTests.java]: Yes, see the note in ElasticsearchException. numbers should never be removed, only assigned to null if the exception is removed.

Comment (jdconrad) [inline core/src/test/java/org/elasticsearch/ExceptionSerializationTests.java]: Got it.  Thanks.

**Comment by jdconrad** [comment]
LGTM.  Glad to see this moved to a plugin.

**Comment by s1monw** [comment]
+1 on moving the tests into the plugin - We can gracefully move them back or to unittests while we go.. I think we should just open a issue that we have to do that so we don't forget! Thanks for working on this!

**Comment by rmuir** [comment]
I just backported this to 2.x

**Pull request merged**